orderer/common/multichannel/registrar.go

```diff
@@ -438,3 +443,66 @@ func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
 	//TODO
 	return errors.New("Not implemented yet")
 }
+
+// validateJoinBlock returns whether this block can be used as a join block for the channel participation API
+// as well as whether the orderer is a member of the consenter set.
+// A join block is a block of either a system channel containing a ConsortiumsConfig, or an application channel
+// containing an ApplicationConfig.
+func (r *Registrar) validateJoinBlock(channelID string, configBlock *cb.Block) (isAppChannel bool, isMember bool, err error) {
+	if !protoutil.IsConfigBlock(configBlock) {
+		return false, false, errors.New("block is not a config block")
+	}
+
+	envelope, err := protoutil.ExtractEnvelope(configBlock, 0)
+	if err != nil {
+		return false, false, err
+	}
+
+	bundle, err := channelconfig.NewBundleFromEnvelope(envelope, r.bccsp)
+	if err != nil {
+		return false, false, err
+	}
+
+	// Check channel id from join block matches
+	if bundle.ConfigtxValidator().ChannelID() != channelID {
```

Comment:
instead of doing this check why can't we just always use the channel ID from within the config block? I mean - to remove the ability to specify a channel ID in the join API altogether and just derive it from the config block

Reply:
because then the channelparticipation api would have to look into the block...

Reply:
Right, so I think it makes more sense to have some kind of abstract validation object which the participation API would delegate the validation to, and it would be done there

Reply:
The link between the channel participation api to the rest of the server state is through the registrar only. As mentioned above, some validation can be performed earlier, but not all.

Reply:
> The link between the channel participation api to the rest of the server state is through the registrar only.

I don't see any harm in either adding another interface to it which will validate join blocks, or just proxying the registrar with a struct that performs validation checks.